File: Python/Permutation_Sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        factor = factorial(n-1)
        k -= 1 # index starts from 1 in the question but our list indexs starts from 0
        ans = []
        numbers_left = list(range(1,n+1))
        
        for m in range(n-1,0,-1):
            index = int(k // factor)
            ans += str(numbers_left[index])
            numbers_left.pop(index)
            k %= factor
            factor /= m
            
        ans += str(numbers_left[0])
        return ''.join(ans)

File: C++/Permutation_Sequence.cpp
// https://leetcode.com/problems/permutation-sequence/
class Solution {
public:
    string getPermutation(int n, int k) {
        vector<bool> chck(n+1,false);
        vector<long long> fac={1,1,2,6,24,120,720,5040,40320,362880};
        string ans;
        int len=0;
        while(len!=n)
        {
            int x=(k-1)/fac[n-len-1];
            k-=fac[n-len-1]*x;
            int count=0;
            for(int i=1;i<=n;i++)
            {
                if(chck[i]==false)
                    count++;
                if(count==(x+1))
                {
                    chck[i]=true;
                    ans+=i+48;
                    len++;
                    break;
                }
            }
        }
        return ans;
    }
};